src/counter_analysis_toolkit/main.c

```diff
@@ -181,6 +181,7 @@ int main(int argc, char*argv[])
     PAPI_shutdown();
 
 #if defined(USE_MPI)
```

Comment:
Is there any communication still going on at this point? If not `MPI_Barrier` should not be needed. If yes can you clarify it?

Reply:
In my testing environment, MPI_Finalize did not block. When a single rank reached MPI_Finalize before others, the others were terminated prematurely.